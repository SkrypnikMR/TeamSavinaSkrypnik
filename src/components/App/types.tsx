import { colors } from '../UI/baseLayout';

export interface ITheme {
    theme: 'dark' | 'light';
    colors: typeof colors;
}

export interface IDiv extends ITheme {
    minHeight: string;
    background: string;
    backgroundSize: string;
    direction: string;
}
