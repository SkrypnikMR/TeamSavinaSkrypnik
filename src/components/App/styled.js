import styled from 'styled-components';

export const StAppDiv = styled.div`
  min-height: 100vh;
  ${({ url = 'heh' }) => url && `background: url(${url}) 100% 100% no-repeat`};
  background-size: cover;
  direction: ${({ lang }) => (lang === 'ar' ? 'rtl' : 'ltr')};
`;
