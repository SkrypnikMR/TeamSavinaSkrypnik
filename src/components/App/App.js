import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';
import { useTranslation } from 'react-i18next';
import Registration from '../Registration';
import Login from '../Login';
import Chat from '../Chat';
import MyAccount from '../MyAccount';
import ModalComponent from '../UI/Modal';
import Component from '../UI/Modal/ModalTestComponent.jsx';
import Header from '../Header';
import { APP_ROUTES } from '/src/constants/reactRoutes';
import { StAppDiv } from './styled';
import ModalInviteUsers from '../UI/Modals/ModalInviteUsers';
import ModalCreateRoom from '../UI/Modals/ModalCreateRoom';

const App = ({ userTheme, userThemeMode }) => {
  const { i18n } = useTranslation();
  return (
    <StAppDiv url={userThemeMode === 'dark' ? userTheme.dark : userTheme.light} lang={i18n.language}>
      <BrowserRouter>
        <Header />
        <Switch>
          <Route path={APP_ROUTES.login} exact component={Login} />
          <Route path={APP_ROUTES.registration} exact component={Registration} />
          <Route path={APP_ROUTES.chat} exact component={Chat} />
          <Route path={APP_ROUTES.account} exact component={MyAccount} />
        </Switch>
      </BrowserRouter>
      <ModalComponent
        Component={Component}
        headerTextKey="notification_settings"
      />
      <ModalCreateRoom />
      <ModalInviteUsers />
      <NotificationContainer />
    </StAppDiv >
  );
};

App.propTypes = {
  userTheme: PropTypes.object,
  userThemeMode: PropTypes.string,
};

export default App;
