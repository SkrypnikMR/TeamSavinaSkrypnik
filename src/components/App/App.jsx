import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';
import { useTranslation } from 'react-i18next';
import { APP_ROUTES } from '../../constants/reactRoutes';
import Registration from '../Registration';
import Login from '../Login';
import ModalComponent from '../UI/Modal';
import Component from '../UI/Modal/ModalTestComponent.jsx';
import { StAppDiv } from './styled';
import MainPage from '../MainPage';

const App = () => {
  const { i18n } = useTranslation();
  return (
    <StAppDiv lang={i18n.language}>
      <BrowserRouter>
        <Switch>
          <Route path={APP_ROUTES.login} exact component={Login} />
          <Route path={APP_ROUTES.registration} exact component={Registration} />
          <Route path={APP_ROUTES.mainPage} exact component={MainPage} />
        </Switch>
      </BrowserRouter>
      <ModalComponent
        Component={Component}
        headerTextKey="notification_settings"
      />
      <NotificationContainer />
    </StAppDiv >
  );
};

export default App;
