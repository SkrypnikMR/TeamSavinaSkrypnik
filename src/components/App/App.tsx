import React from 'react';
import { Route, Switch, BrowserRouter } from 'react-router-dom';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';
import { useTranslation } from 'react-i18next';

import { APP_ROUTES } from '../../constants/reactRoutes';
import Registration from '../Registration/Registration';
import Login from '../Login';
import MainPage from '../MainPage';
import StatisticPage from '../StatisticPage';
import { useTheme } from '../Hook/useTheme';

import { StAppDiv } from './styled';

const App = () => {
  const { i18n } = useTranslation();
  const { colors, theme } = useTheme();
  return (
    <StAppDiv lang={i18n.language} colors={colors} theme={theme}>
      <BrowserRouter>
        <Switch>
          <Route path={APP_ROUTES.login} exact component={Login} />
          <Route path={APP_ROUTES.registration} exact component={Registration} />
          <Route path={APP_ROUTES.mainPage} exact component={MainPage} />
          <Route path={APP_ROUTES.statistics} exact component={StatisticPage} />
        </Switch>
      </BrowserRouter>
      <NotificationContainer />
    </StAppDiv >
  );
};

export default App;
