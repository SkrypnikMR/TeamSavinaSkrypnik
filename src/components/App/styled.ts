import styled from 'styled-components';

import { IDiv } from './types';

export const StAppDiv = styled.div < IDiv > `
  min-height: 100vh;
  background: ${({ colors, theme }) => colors[theme].background};
  background-size: cover;
`;
