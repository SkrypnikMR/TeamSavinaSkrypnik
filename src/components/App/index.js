import App from './App';
import { connect } from 'react-redux';
import { userTheme, userThemeMode } from '/src/Store/user/selectors';

const mapStateToProps = state => ({
  userTheme: userTheme(state),
  userThemeMode: userThemeMode(state),
});

export default connect(mapStateToProps)(App);
