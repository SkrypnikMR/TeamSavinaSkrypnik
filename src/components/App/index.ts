import App from './App';
import { withTheme } from '../HOC';

export default withTheme(App);
