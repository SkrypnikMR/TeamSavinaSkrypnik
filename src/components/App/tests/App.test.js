import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart } from '/src/helpers/testHelper';
import App from '../App';
import '/src/i18n';

jest.mock('react-modal');

const mockStore = configureStore();
const store = mockStore({
  registration: {
    email: '',
    password: '',
    confirm: '',
    firstName: '',
    lastName: '',
    success: null,
  },
  login: {
    email: '',
    password: '',
    confirm: '',
    firstName: '',
    lastName: '',
    success: null,
  },
  user: {
    themeMode: 'light',
  },
  modals: {
    notificationSettings: {
        type: 'notificationSettings',
        data: {},
        isOpen: true,
    },
    logOut: {
        type: 'logOut',
        data: {},
        isOpen: false,
    },
    allNotification: {
        type: 'allNotification',
        data: {},
        isOpen: false,
    },
    usersInChat: {
        type: 'usersInChat',
        data: {},
        isOpen: false,
    },
    createChat: {
        type: 'usersInChat',
        data: {},
        isOpen: false,
    },
  },
});
const props = { 
  userTheme: {
    light: 'lightBackground',
    dark: 'darkBackground',
  },
  userThemeMode: 'dark',
};

describe('App', () => {
  it('Should match snapshot', () => {
    const component = shallowSmart(<App {...props} />, store);
    expect(component.html()).toMatchSnapshot();
  });
});
