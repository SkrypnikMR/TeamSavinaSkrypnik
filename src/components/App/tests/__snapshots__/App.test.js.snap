// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `"<div lang=\\"en\\" class=\\"styled__StAppDiv-sc-7r44pg-0 jraHDu\\"><div class=\\"styled__StHeader-zzrvsn-0 hDaupM\\"><div class=\\"styled__StLogo-sc-9vy3sh-0 dOfUuH\\"><img src=\\"./public/assets/images/logo.png\\"/><h1>Header&amp;Shmeader</h1></div><div class=\\"styled__StControl-sc-1i99xqk-0 joBAxQ\\"><button id=\\"theme_btn\\" type=\\"button\\" value=\\"dark\\" color=\\"white\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 jFhuDa\\">ðŸŒ—</button><button id=\\"en\\" type=\\"button\\" value=\\"en\\" color=\\"white\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 jFhuDa\\">EN</button><button id=\\"ru\\" type=\\"button\\" value=\\"ru\\" color=\\"white\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 jFhuDa\\">RU</button><button id=\\"ar\\" type=\\"button\\" value=\\"ar\\" color=\\"white\\" font-size=\\"26px\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 jFhuDa\\">AR</button></div></div><div class=\\"styled__StLoginDiv-ohf2t6-0 cCvQFI\\"><div class=\\"styled__StFormDiv-sc-1xa63f-0 hNLlQX\\"><p>Sign In</p><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"email\\" class=\\"styled__StLabel-w3ykst-0 fbdCHZ\\">Email</label><input type=\\"text\\" id=\\"email\\" name=\\"email\\" value=\\"\\" placeholder=\\"Enter your email...\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"password\\" class=\\"styled__StLabel-w3ykst-0 fbdCHZ\\">Password</label><input type=\\"text\\" id=\\"password\\" name=\\"password\\" value=\\"\\" placeholder=\\"Enter your password...\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><button id=\\"login\\" type=\\"button\\" height=\\"50px\\" class=\\"styled__StButton-sc-1a6hck8-0 euTlOK\\">Login</button><span>Need an account?<a href=\\"/registration\\">Register now</a></span></div></div><div class=\\"notification-container notification-container-empty\\"><div></div></div></div>"`;
